### PR TITLE
Create GitHub releases for merges into main

### PR DESCRIPTION
**Pull request by JoeRobich**
```mermaid
sequenceDiagram
  autonumber
  Maintainer ->> GitHub: Merges PR into the `master` branch
  GitHub --) OmniSharp ADO: Merge to `master` triggers pipeline
  activate OmniSharp ADO
  OmniSharp ADO ->> GitHub: Pulls source for omnisharp-roslyn
  note over OmniSharp ADO: Calculates a build version
  OmniSharp ADO ->> GitHub: Creates a draft release and `v#35;.#35;.#35;-beta.#35;` tag
  deactivate OmniSharp ADO
  GitHub --) OmniSharp ADO: `v*` tag creation triggers pipeline
  activate OmniSharp ADO
  OmniSharp ADO ->> GitHub: Pulls source for omnisharp-roslyn
  note over OmniSharp ADO: Builds packages for various platforms
  OmniSharp ADO ->> GitHub: Adds packages to release and unmark as draft
  deactivate OmniSharp ADO
```

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -42,152 +27,183 @@ variables:
...
-      #     allowPackageConflicts: true
-      #   condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags/v'))
-      - script: |
-          AZ_REPO=$(lsb_release -cs)
-          echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $AZ_REPO main" | \
-            sudo tee /etc/apt/sources.list.d/azure-cli.list
-          curl -L https://packages.microsoft.com/keys/microsoft.asc | sudo apt-key add -
-          sudo apt-get install apt-transport-https
-          sudo apt-get update && sudo apt-get install azure-cli
-          az storage blob upload-batch --destination "releases" --source "$(System.ArtifactsDirectory)/deployment/" --overwrite --account-name roslynomnisharp --account-key $BLOB_KEY
```
Comment (JoeRobich) [inline azure-pipelines.yml:173]: We will no longer push artifacts to this storage account.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -42,152 +27,183 @@ variables:
...
-      - task: NuGetCommand@2
-        displayName: "Push NuGet packages to Azure Artifacts"
-        inputs:
-          command: push
-          nuGetFeedType: internal
-          packagesToPush: "$(System.ArtifactsDirectory)/nuget/**/*.nupkg"
-          publishVstsFeed: Builds/omnisharp
-          allowPackageConflicts: true
-        condition: and(succeeded(), or(startsWith(variables['Build.SourceBranch'], 'refs/tags/v'), eq(variables['Build.SourceBranch'], 'refs/heads/master')))
-      # Temporarily disable publishing to nuget.org while we attempt to renew authentication.
```
Comment (JoeRobich) [inline azure-pipelines.yml:156]: Reenabled pushing to NuGet for official builds.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -42,152 +27,183 @@ variables:
...
   VstsArtifacts: "$(Artifacts)"
   DOTNET_SKIP_FIRST_TIME_EXPERIENCE: "true"
+  IsMaster: $[eq(variables['Build.SourceBranch'], 'refs/heads/master')]
+  IsRelease: $[startsWith(variables['Build.SourceBranch'], 'refs/tags/v')]
+  IsBetaRelease: $[and(startsWith(variables['Build.SourceBranch'], 'refs/tags/v'), contains(variables['Build.SourceBranch'], '-beta.'))]
+  IsOfficialRelease: $[and(startsWith(variables['Build.SourceBranch'], 'refs/tags/v'), not(contains(variables['Build.SourceBranch'], '-beta.')))]
+
+stages:
+  - stage: CreateBetaReleaseTag
+    displayName: "Tag Beta Release"
```
Comment (JoeRobich) [inline azure-pipelines.yml:37]: Running against master branch no longer builds artifacts directly. Instead it creates a draft release and tag which will then cause this pipeline to build a prerelease.

File: azure-pipelines.yml [hunk 4]
```diff
@@ -1,26 +1,11 @@
...
-pr:
-  autoCancel: "true"
-  branches:
-    include:
-      - master
-      - feature/*
-  paths:
-    exclude:
-      - 'CHANGELOG.md'
+pr: none
```
Comment (JoeRobich) [inline azure-pipelines.yml:8]: We already have a "build" pipeline in GH actions which performs the same steps that this pipeline did for PRs.

File: azure-pipelines.yml [hunk 5]
```diff
@@ -1,26 +1,11 @@
-parameters:
-  - name: testProjects
```
Comment (JoeRobich) [inline azure-pipelines.yml:2]: Removed because this pipeline hasn't run unit tests in years.

File: azure-pipelines.yml (hunk 4 above)
Comment (JoeRobich) [reply azure-pipelines.yml:8]: Branch protections rules have been updated accordingly.

**Review by 333fred — APPROVED** [review]

**Pull request merged**